### PR TITLE
vm_factory: check network setup

### PR DESCRIPTION
**Pull request by bergwolf**
To make sure that we have properly configured network endpoints for
vm factory.

Fixes: #874
Depends-on: github.com/kata-containers/runtime#872

Signed-off-by: Peng Tao <bergwolf@gmail.com>

### CODE REVIEW

**Comment by bergwolf** [comment]
CI failed on various errors all non-related to this patch:
1. http://jenkins.katacontainers.io/job/kata-containers-tests-ubuntu-16-04-PR-initrd/590/console
```
 Failure in Spec Setup (BeforeEach) [1.092 seconds]
run
/tmp/jenkins/workspace/kata-containers-tests-ubuntu-16-04-PR-initrd/go/src/github.com/kata-containers/tests/functional/run_test.go:20
  container [BeforeEach]
  /tmp/jenkins/workspace/kata-containers-tests-ubuntu-16-04-PR-initrd/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
    without '--bundle' option
    /tmp/jenkins/workspace/kata-containers-tests-ubuntu-16-04-PR-initrd/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46

    Expected error:
        <*exec.ExitError | 0xc00000c5c0>: {
            ProcessState: {
                pid: 32231,
                status: 256,
                rusage: {
                    Utime: {Sec: 0, Usec: 112000},
                    Stime: {Sec: 0, Usec: 20000},
                    Maxrss: 31408,
                    Ixrss: 0,
                    Idrss: 0,
                    Isrss: 0,
                    Minflt: 3053,
                    Majflt: 210,
                    Nswap: 0,
                    Inblock: 50536,
                    Oublock: 0,
                    Msgsnd: 0,
                    Msgrcv: 0,
                    Nsignals: 0,
                    Nvcsw: 518,
                    Nivcsw: 49,
                },
            },
            Stderr: nil,
        }
        exit status 1
    not to have occurred

    /tmp/jenkins/workspace/kata-containers-tests-ubuntu-16-04-PR-initrd/go/src/github.com/kata-containers/tests/functional/run_test.go:28
```
2. http://jenkins.katacontainers.io/job/kata-containers-tests-vsock-fedora-27-PR/367/console
```
time="2018-10-30T10:29:38Z" level=error msg="error copying docker://redis:alpine to dir:/tmp/jenkins/workspace/kata-containers-tests-vsock-fedora-27-PR/go/src/github.com/kubernetes-incubator/cri-o/.artifacts/redis-image: Error determining manifest MIME type for docker://redis:alpine: pinging docker registry returned: error pinging repository, response code 503" 
Error pulling docker://redis
make: *** [Makefile:56: crio] Error 1
Build step 'Execute shell' marked build as failure
Performing Post build task...
Match found for :.* : True
Logical operation result is TRUE
Running script  : #!/bin/bash
```
3. http://jenkins.katacontainers.io/job/kata-containers-tests-fedora-27-PR/650/consoleFull
```
Running command '/usr/bin/docker [docker pull postgres]'
command failed error 'exit status 1'
[docker pull postgres]
Timeout: 600 seconds
Exit Code: 1
Stdout: Using default tag: latest
```
4. http://jenkins.katacontainers.io/job/kata-containers-tests-cri-containerd-PR/99/console
```
go test -c github.com/containerd/cri/integration -o _output/integration.test
=== RUN   TestImageLoad
--- FAIL: TestImageLoad (0.75s)
    image_load_test.go:41: docker save image into tarball
    assertions.go:229: 
                        
	Error Trace:	image_load_test.go:43
        
	Error:		Received unexpected error exit status 1
        
	Messages:	output: "Error response from daemon: unauthorized: authentication required\n"
        
FAIL
./hack/test-utils.sh: line 63: 30366 Terminated              keepalive "sudo PATH=${PATH} ${ROOT}/_output/containerd ${CONTAINERD_FLAGS}" ${RESTART_WAIT_PERIOD} &> ${report_dir}/containerd.log
```

**Comment by marcov** [comment]
lgtm

File: integration/vm_factory/vm_templating_test.sh [hunk 1]
```diff
@@ -65,9 +65,15 @@ check_qemu_for_vm_template() {
 	[ $? -eq 0 ] || die "vm is not backed by vm template"
 }
 
+check_vm_template_network_setup() {
+	IPADDR=$(sudo docker exec -t $CONTAINER_NAME ip addr show eth0| sed -ne "s|.*inet \(.*\)/.* brd .*|\1|p")
```
Comment (marcov) [inline integration/vm_factory/vm_templating_test.sh]: just fyi you could use `ip -4 -br a show eth0` that's easier to parse :)

**Comment by chavafg** [comment]
lgtm

**Pull request merged**